File: util/b2m-sms/src/main/java/top/bettercode/sms/b2m/B2mResponse.java
package top.bettercode.sms.b2m;

import java.util.List;
import top.bettercode.lang.property.PropertiesSource;

/**
 * @author Peter Wu
 */
public class B2mResponse {

  public static final String SUCCESS="SUCCESS";
  private static final PropertiesSource codeMessageSource = PropertiesSource.of("b2m-message");
  private String code;

  private List<B2mRespData> data;

  public static String getMessage(String code) {
    return codeMessageSource.getOrDefault(code, code);
  }

  public String getCode() {
    return code;
  }

  public void setCode(String code) {
    this.code = code;
  }

  public List<B2mRespData> getData() {
    return data;
  }

  public void setData(List<B2mRespData> data) {
    this.data = data;
  }

  public boolean isOk() {
    return SUCCESS.equals(code);
  }

  public String getMessage() {
    return codeMessageSource.getOrDefault(code, code);
  }
}

File: clients/sms/src/main/kotlin/top/bettercode/summer/tools/sms/b2m/B2mResponse.kt
package top.bettercode.summer.tools.sms.b2m

import top.bettercode.summer.tools.lang.client.ClientResponse
import top.bettercode.summer.tools.lang.property.PropertiesSource.Companion.of

/**
 * @author Peter Wu
 */
class B2mResponse<T> : ClientResponse {
    //--------------------------------------------
    var code: String? = null
    var data: List<T>? = null

    constructor()
    constructor(data: List<T>?) {
        this.code = SUCCESS
        this.data = data
    }

    override val isOk: Boolean
        //--------------------------------------------
        get() = SUCCESS == code
    override val message: String
        get() = codeMessageSource.getOrDefault(code!!, code!!)

    companion object {
        const val SUCCESS = "SUCCESS"
        private val codeMessageSource = of("b2m-message")

        //--------------------------------------------
        fun getMessage(code: String?): String? {
            return code?.let { codeMessageSource.getOrDefault(code, it) }
        }
    }
}
